hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -5346,11 +5346,12 @@ NamenodeCommand startCheckpoint(NamenodeRegistration backupNode,
   public void processIncrementalBlockReport(final DatanodeID nodeID,
       final StorageReceivedDeletedBlocks srdb)
       throws IOException {
-    writeLock();
+    // Needs the FSWriteLock since it may update quota and access storage policyId and full path.
```

Comment:
means both fs and bm locks are needed here?

Reply:
Yes. `completeBlock` will updateQuota, so it needs BMWriteLock and FSWriteLock.

`processExtraRedundancyBlock` chooses excess replica depends on storage policyId, so it needs FSReadLock.

`isInSnapshot` depends on the full path, so it needs FSReadLock.

Reply:
Better to make the comment clear.
FS lock in the comment, but global lock in codes.